src/stcal/ramp_fitting/ols_fit.py

```diff
@@ -1366,6 +1365,12 @@ def ramp_fit_overall(
     #   primary output
     final_pixeldq = utils.dq_compress_final(dq_int, ramp_data)
 
+    # For invalid slope calculations set to NaN.  Pixels flagged as SATURATED or
+    # DO_NOT_USE have invalid data.
+    invalid_data = ramp_data.flags_saturated | ramp_data.flags_do_not_use
```

Comment:
I assume this is applied *after* the point at which saturation flags are weeded out for pixels that are only partially saturated?  In other words, at this point a pixel will only have the saturation flag set if all groups are saturated, right?

Reply:
Yes.  You can see in the lines above that the final DQ computations are done in `final_pixeldq`.  After the final DQ computations are done, those flags are then used to identify invalid slops computed, which at this line in the code is represented by `c_rates`.  This computation on `c_rates` is the last computation done before returning it as the computed slopes.